plugins/telemetry-otel/src/main/java/org/opensearch/telemetry/tracing/sampler/ProbabilisticSampler.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.telemetry.tracing.sampler;
+
+import org.opensearch.telemetry.TelemetrySettings;
+
+import java.util.List;
+import java.util.Objects;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.trace.data.LinkData;
+import io.opentelemetry.sdk.trace.samplers.Sampler;
+import io.opentelemetry.sdk.trace.samplers.SamplingResult;
+
+/**
+ * ProbabilisticSampler implements a head-based sampling strategy based on provided settings.
+ */
+public class ProbabilisticSampler implements Sampler {
+    private Sampler defaultSampler;
+    private final TelemetrySettings telemetrySettings;
+    private double tracerHeadSamplerSamplingRatio;
+
+    /**
+     * Constructor
+     *
+     * @param telemetrySettings Telemetry settings.
+     */
+    public ProbabilisticSampler(TelemetrySettings telemetrySettings) {
+        Objects.requireNonNull(telemetrySettings);
+        this.telemetrySettings = telemetrySettings;
+        this.tracerHeadSamplerSamplingRatio = telemetrySettings.getTracerHeadSamplerSamplingRatio();
+        this.defaultSampler = createProbabilitySampler(tracerHeadSamplerSamplingRatio);
+    }
+
+    private Sampler createProbabilitySampler(double samplingRatio) {
```

Comment:
Why we need this method indirection to call another another method? Why not to use `Sampler.traceIdRatioBased(samplingRatio)` directly?

Reply:
Actually we were calling `createProbabilitySampler` at 2 places in same class. But it makes sense to call `Sampler.traceIdRatioBased` directly instead of creating another function for it.

I have made changes accordingly in the latest commit.